pkg/controller/job/job_controller.go

```diff
@@ -1364,11 +1380,21 @@ func jobSuspended(job *batch.Job) bool {
 	return job.Spec.Suspend != nil && *job.Spec.Suspend
 }
 
+// Recreate pods once they have fully transitioned to failed
+// This means that terminating pods will wait until they are fully deleted.
+func recreatePodsWhenFailed(job *batch.Job) bool {
+	if !feature.DefaultFeatureGate.Enabled(features.JobPodReplacementPolicy) {
```

Comment:
Ultimately we should apply this semantic also when JobPodFailurePolicy is used as planned in the KEP for Third Beta: https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/3329-retriable-and-non-retriable-failures#beta. 

Do you have a plan to update it in this PR, or a dedicated one? It seems a small change so maybe it makes sense to roll it here as we did for the KEP update. What is your view @alculquicondor ?

Reply:
Thank you to you and @alculquicondor for pointing this out in the KEP.  I missed this as a case to add.